lib/widgets/editor/tree_widgets/editor_settings_tree.dart

```diff
@@ -123,4 +109,54 @@ class _EditorSettingsTreeState extends State<EditorSettingsTree> {
       ],
     );
   }
+
+  Widget _buildCheckboxRow(
+      String label, bool value, Function(bool?) onChanged, String tooltip) {
+    return Row(
+      children: [
+        Tooltip(
+          message: tooltip,
+          child: Checkbox(
+            value: value,
+            onChanged: onChanged,
+          ),
+        ),
+        Padding(
+          padding: const EdgeInsets.only(
+            bottom: 3.0,
+            left: 4.0,
+          ),
+          child: Text(
+            label,
+            style: const TextStyle(fontSize: 15),
+          ),
+        ),
+      ],
+    );
+  }
+
+  void _updateSetting(String key, bool? value) {
+    if (value != null) {
+      setState(() {
+        switch (key) {
+          case PrefsKeys.snapToGuidelines:
+            _snapToGuidelines = value;
+            break;
+          case PrefsKeys.hidePathsOnHover:
+            _hidePathsOnHover = value;
+            break;
+          case PrefsKeys.showStates:
+            _showStates = value;
+            break;
+          case PrefsKeys.showRobotDetails:
+            _showRobotDetails = value;
+            break;
+          case PrefsKeys.showGrid:
+            _showGrid = value;
+            break;
+        }
+        _prefs.setBool(key, value);
+      });
+    }
+  }
```

Comment:
Not really a fan of using a switch statement like this here. I'd probably just replace every usage of this method with something like
```
setState(() {
    _snapToGuidelines = value!;
});
_prefs.setBool(PrefsKeys.snapToGuidelines, _snapToGuidelines);
```
You can safely use the `!` operator since these are not 3 state checkboxes and will never return null. Just makes things a bit cleaner and easier to read.

Reply:
Yeah that would make it quite easier to read